src/routes/ArticleDetailPage.jsx

```diff
@@ -1,7 +1,54 @@
-import React from "react";
+import React, { useContext, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { ArticlesContext } from "../context/ArticlesContext";
+import ArticleFinder from "../apis/ArticleFinder";
 
 const ArticleDetailPage = () => {
-  return <div>Article Detail Page</div>;
+  const { id } = useParams();
+  const { selectedArticle, setSelectedArticle } = useContext(ArticlesContext);
```

Comment:
Hi Modou, initially I thought this would be a good way to avoid prop drilling and improve scalability by avoiding lots of re-fetching but now I see what you mean about the  article loading and the old one still being present for a second. I think I will have another go at refactoring this to set state for each page as this is only a small application.

Reply:
My refactor is now complete